ibm/service/cos/resource_ibm_cos_replication_configuration.go

```diff
@@ -0,0 +1,438 @@
+
+			if err != nil {
+				return fmt.Errorf("failed to update the replication rule on COS bucket %s, %v", bucketName, err)
+			}
+		}
+	}
+	return resourceIBMCOSBucketReplicationConfigurationRead(d, meta)
+}
+
+func resourceIBMCOSBucketReplicationConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+
+	bucketCRN := parseBucketReplId(d.Id(), "bucketCRN")
+	bucketName := parseBucketReplId(d.Id(), "bucketName")
+	bucketLocation := parseBucketReplId(d.Id(), "bucketLocation")
+	instanceCRN := parseBucketReplId(d.Id(), "instanceCRN")
+	endpointType := parseBucketReplId(d.Id(), "endpointType")
+
+	d.Set("bucket_crn", bucketCRN)
+	d.Set("bucket_location", bucketLocation)
+	if endpointType != "" {
+		d.Set("endpoint_type", endpointType)
+	}
+
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+	if err != nil {
+		return err
+	}
+
+	getBucketReplicationInput := &s3.GetBucketReplicationInput{
+		Bucket: aws.String(bucketName),
+	}
+
+	replicationptr, err := s3Client.GetBucketReplication(getBucketReplicationInput)
+
+	if err != nil && !strings.Contains(err.Error(), "AccessDenied: Access Denied") {
+		return err
+	}
+
+	if replicationptr != nil {
+		replicationRules := flex.ReplicationRuleGet(replicationptr.ReplicationConfiguration)
+		if len(replicationRules) > 0 {
+			d.Set("replication_rule", replicationRules)
+		}
+	}
+
+	return nil
+}
+
+func resourceIBMCOSBucketReplicationConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	bucketName := parseBucketReplId(d.Id(), "bucketName")
+	bucketLocation := parseBucketReplId(d.Id(), "bucketLocation")
+	instanceCRN := parseBucketReplId(d.Id(), "instanceCRN")
+	endpointType := parseBucketReplId(d.Id(), "endpointType")
+
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+	if err != nil {
+		return err
+	}
+
+	deleteBucketReplicationInput := &s3.DeleteBucketReplicationInput{
+		Bucket: aws.String(bucketName),
+	}
+
+	_, err = s3Client.DeleteBucketReplication(deleteBucketReplicationInput)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func parseBucketReplId(id string, info string) string {
+	bucketCRN := strings.Split(id, ":meta:")[0]
+	meta := strings.Split(id, ":meta:")[1]
+
+	if info == "bucketName" {
+		return strings.Split(bucketCRN, ":bucket:")[1]
+	}
+	if info == "instanceCRN" {
+		return fmt.Sprintf("%s::", strings.Split(bucketCRN, ":bucket:")[0])
+	}
+	if info == "bucketCRN" {
+		return bucketCRN
+	}
+	if info == "bucketLocation" {
+		return strings.Split(meta, ":")[0]
+	}
+	if info == "endpointType" {
+		return strings.Split(meta, ":")[1]
+	}
+
+	return parseBucketId(bucketCRN, info)
+}
+
+func getCosEndpointType(bucketLocation string, endpointType string) string {
```

Comment:
why are you definig this method can't we use the old exisiting methods

Reply:
it will be overwrite ...thatswhy created new one